pkg/kubelet/kuberuntime/helpers_test.go

```diff
@@ -31,6 +31,56 @@ import (
 	utilpointer "k8s.io/utils/pointer"
 )
 
+func TestIsInitContainerFailed(t *testing.T) {
+	tests := []struct {
+		status   *kubecontainer.Status
+		isFailed bool
+	}{
+		{
+			status: &kubecontainer.Status{
+				State:    kubecontainer.ContainerStateExited,
+				ExitCode: 1,
+			},
+			isFailed: true,
+		},
+		{
+			status: &kubecontainer.Status{
+				State: kubecontainer.ContainerStateUnknown,
+			},
+			isFailed: true,
+		},
+		{
+			status: &kubecontainer.Status{
```

Comment:
The scenario solved by this pr is that exit code is 0 and the reason is OOMKilled. I think you need to add some test cases for this scenario.

Reply:
yes, I think so,  I will add it.